evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,373 @@
+                slider.find("[data-action=repeat-unreviewed]").toggleClass("d-none", reviewed);
+                slider.find("[data-action=repeat-all]").toggleClass("d-none", items.length === 0);
+                slideLayer(2, direction, alert);
+            } else {
+                slideLayer(2, direction, items.eq(index));
+            }
+
+            slider.find(".slider-side-left").toggleClass("shown", index > 0);
+            let reviewed = items.slice(0, index).filter("[data-review]").length;
+            slider.find("[data-counter=reviewed-left]").text(reviewed);
+            slider.find("[data-counter=unreviewed-left]").text(index - reviewed);
+
+            slider.find(".slider-side-right").toggleClass("shown", index < items.length);
+            reviewed = items.slice(index + 1).filter("[data-review]").length;
+            slider.find("[data-counter=reviewed-right]").text(reviewed);
+            let unreviewed = items.length - index - reviewed - 1;
+            slider.find("[data-counter=unreviewed-right]").text(Math.max(unreviewed, 0));
+        }
+
+        function slideLayer(layer, direction, element) {
+            // to preserve the vertical positions and heights during transition,
+            // the elements will be deactivated from bottom to top and activated from top to bottom
+
+            if(element && element.is(".active")) {
+                return;
+            }
+
+            let last = findActive(layer);
+            if(last.length !== 0) {
+                let reversed = direction === "left"? "right" : "left";
+                last.css({top: last.position().top, height: last.outerHeight()});
+                last.removeClass("active to-left to-right");
+                last.addClass("to-" + reversed);
+            }
+
+            if(layer > 0) {
+                if(index < items.length) {
+                    let reference = element.prevAll("[data-layer=" + (layer - 1) + "]").first();
+                    slideLayer(layer - 1, direction, reference);
+                } else {
+                    slideLayer(layer - 1, direction, null);
+                }
+            }
+
+            if(element !== null) {
+                element.removeClass("to-left to-right");
+                element.addClass("to-" + direction);
+                element.position();
+                element.addClass("active");
+            }
+        }
+
+        function findActive(layer) {
+            if(layer === 2) {
+                return slider.find("[data-layer=2].active, .alert.active");
+            } else {
+                return slider.find("[data-layer=" + layer + "].active");
+            }
+        }
+
+        function repeat(action) {
+            let reviewed = slider.find("[data-layer=2][data-review]");
+            let unreviewed = slider.find("[data-layer=2]:not([data-review])");
+            let index = action === "unreviewed" && unreviewed.length? reviewed.length : 0;
```

Comment:
Yes, the `index` must be assigned before `merge()`, as it will modify the first argument and return the modified array. I played around with other options but this looked like the cleanest to me.

Just noted after push that `index` can be renamed to `to` in this scope.